wrappers/services.go

```diff
@@ -191,6 +191,7 @@ Requires=snapd.frameworks.target{{ if .App.Socket }} {{.SocketFileName}}{{end}}
 X-Snappy=yes
 
 [Service]
+ExecPreStart=mkdir -p {{.App.Snap.RootUserDataDir}} {{.App.Snap.RootCommonUserDataDir}}
```

Comment:
This is a general problem we have for any snaps, no matter the user or whether it's a service or not. The solution we agreed to back then was to have snap-exec creating it.

Can we have that instead?